numpy/linalg/_linalg.py

```diff
@@ -2011,6 +2011,7 @@ def cond(x, p=None):
         # contain nans in the entries where inversion failed.
         _assert_stacked_square(x)
         t, result_t = _commonType(x)
+        result_t = _realType(result_t)  # condition number is always real
         signature = 'D->D' if isComplexType(t) else 'd->d'
         with errstate(all='ignore'):
             invx = _umath_linalg.inv(x, signature=signature)
```

Comment:
So, I think this is fine.  Your other comment (deleted now) is not wrong, in that it would be fine to cast `invx` to the original `result_t` to achieve the right thing.
(Norm might in some paths end up casting back and forth then, but it would work too.)